nifi-nar-bundles/nifi-media-bundle/nifi-media-processors/src/test/java/org/apache/nifi/processors/document/ExtractDocumentTextTest.java

```diff
@@ -0,0 +1,256 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.document;
+
+import org.apache.nifi.util.MockFlowFile;
+import org.apache.nifi.util.TestRunner;
+import org.apache.nifi.util.TestRunners;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.UnsupportedEncodingException;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+
+public class ExtractDocumentTextTest {
+
+    private TestRunner testRunner;
+
+    @BeforeEach
+    public void init() {
+        testRunner = TestRunners.newTestRunner(ExtractDocumentText.class);
+    }
+
+    @Test
+    public void processor_should_support_pdf_types_without_exception() {
+        try {
+            final String filename = "simple.pdf";
+            MockFlowFile flowFile = testRunner.enqueue(new FileInputStream("src/test/resources/" + filename));
+            Map<String, String> attrs = new HashMap<String, String>() {{ put("filename", filename); }};
+            flowFile.putAttributes(attrs);
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
+
+        testRunner.assertValid();
+        testRunner.run();
+        testRunner.assertTransferCount(ExtractDocumentText.REL_FAILURE, 0);
+
+        List<MockFlowFile> successFiles = testRunner.getFlowFilesForRelationship(ExtractDocumentText.REL_SUCCESS);
+        for (MockFlowFile mockFile : successFiles) {
+            try {
+                String result = new String(mockFile.toByteArray(), "UTF-8");
+                String trimmedResult = result.trim();
+                assertTrue(trimmedResult.startsWith("A Simple PDF File"));
+                System.out.println("FILE:" + result);
+            } catch (UnsupportedEncodingException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
+    @Test
+    public void processor_should_support_doc_types_without_exception() {
+        try {
+            final String filename = "simple.doc";
```

Comment:
Having multiple sample binary files adds unnecessary content to the source code repository.  Apache Tika supports a wide array of formats, so having multiple types of documents does not provide different processing in the NiFi code itself.  For this reason, recommend removing all binary files and instead testing with some simple strings.

Reply:
I'm going to skip this one. I think this provides a really good and cheap way to do a meaningful functional test of the processor for anyone who has to step in and fix it or update it later.

Reply:
That's fair, perhaps reducing to a single binary document would be a sufficient way to go.